### ISSUE TITLE
Index files stay on disk after the index was dropped

### ISSUE THREAD

**Issue by puzpuzpuz** [questdb/questdb#5141]
### To reproduce

```sql
create table x (ts timestamp, id symbol) timestamp(ts) partition by day;
insert into x values (now(), '1');

alter table x alter column id add index;
alter table x alter column id drop index;
```

Index files (`.k` and `.v` files from the partition dirs) stay on the disk while they should be deleted eventually.

### QuestDB version:

latest master

### OS, in case of Docker specify Docker and the Host OS:

Linux

### File System, in case of Docker specify Host File System:

ext4

### Full Name:

Andrei Pechkurov

### Affiliation:

QuestDB

### Have you followed Linux, MacOs kernel configuration steps to increase Maximum open files and Maximum virtual memory areas limit?

- [X] Yes, I have

**Comment by puzpuzpuz** [comment]
> 1. When I remove index column it changes `id.d` -> `id.d.1`

Yes, that's because we create a new version of the column. `*.d` files stand for the data column files.

> 2. Is this the class where I need to add logic for removing index files

The easiest way to see how the code works is to set a few break points in `TableWriter#dropIndex(CharSequence)` and debug through the drop index SQL execution.

**Comment by ideoma** [comment]
Fixed in #5242
